src/deposit.cairo

```diff
@@ -85,37 +88,71 @@ mod Deposit {
         PositionClosed: PositionClosed
     }
 
-    #[abi(embed_v0)]
-    impl Deposit of IDeposit<ContractState> {
+    #[generate_trait]
+    impl InternalImpl of InternalTrait {
         fn swap(ref self: ContractState, swap_data: SwapData) -> SwapResult {
             ekubo::components::shared_locker::call_core_with_callback(
                 self.ekubo_core.read(), @swap_data
             )
         }
+    }
 
+    #[abi(embed_v0)]
+    // TODO: Change types to aliases
+    impl Deposit of IDeposit<ContractState> {
+        /// Loops collateral token on ZKlend.
+        ///
+        /// # Panics
+        ///
+        /// `is_position_open` storage variable is set to true('Open position already exists')
+        /// `amount` field of `deposit_data` is `0` or `pool_price` is `0`
+        /// address of the caller is not equal to `owner` storage address
+        ///
+        /// # Paraemters
+        /// * `deposit_data` - Object of which stores main deposit information.
+        /// * `pool_key` - Ekubo type which represents data about pool.
+        /// * `pool_price` - Price of `deposit` token in terms of `debt` token in Ekubo pool.
+        /// * `usdc_price` - Price of `deposit` token in USDC.
         fn loop_liquidity(
             ref self: ContractState,
             deposit_data: DepositData,
             pool_key: PoolKey,
-            pool_price: felt252
+            pool_price: u256,
+            usdc_price: u256
         ) {
-            assert(get_caller_address() == self.owner.read(), 'Caller is not an owner');
+            // TODO: Add borrow factor
+            let user_acount = get_tx_info().unbox().account_contract_address;
+            assert(user_acount == self.owner.read(), 'Caller is not the owner');
             assert(!self.is_position_open.read(), 'Open position already exists');
             let DepositData { token, amount, multiplier } = deposit_data;
+            let token_dispatcher = ERC20ABIDispatcher { contract_address: token };
+            let deposit_token_decimals = fast_power(10_u128, token_dispatcher.decimals().into());
+
+            assert(multiplier < 5 && multiplier > 1, 'Multiplier not supported');
             assert(amount != 0 && pool_price != 0, 'Parameters cannot be zero');
-            assert(multiplier < 5, 'Multiplier not supported');
+            assert(
+                amount * usdc_price / deposit_token_decimals.into() >= 1000000,
+                'Loop amount is too small'
+            ); // User needs to have at least 1 USDC, so rounded down amount is OK.
```

Comment:
You are only using usdc_price to calculate whether the loan amount is at least 1 USDC.

1. Why is this limitation in place?
2. This does not actually check anything because the caller can spoof the usdc_price. (!!)
3. If this is really necessary (and I don't think it is), you can adjust the usdc_price format such that you _don't need_ to fetch decimals! (Basically embed the decimals there and pass only the ratio) 
4. usdc_price is actually not really price of USDC, is it?

Reply:
I added it because sometimes for small amounts depositing and borrowing is failing because of number of iterations(or steps made by program). I can remove it but sometimes transaction will fail with unclear message(same will be if user or our frontend inputs wrong price). I had some doubts when adding this because of the same reason, so I will remove it and try to locate the reason and location of fail, it's number of iterations I would consider limiting it instead(number of deposits and borrowings in a loop).